kuma/javascript/src/index.jsx

```diff
@@ -34,4 +34,19 @@ if (container) {
         // server side rendering.
         ReactDOM.render(app, container);
     }
+
+    // Finally, here is some other miscellaneous code that we need to run.
+    // The idea is that this is code need to make the non-React parts of
+    // the UX work correctly. If we end up with more than a couple of things
+    // here we should refactor them into a separate module
+
+    // An event handler to make the language selector menu
+    // in the footer work correctly.
+    // See also kuma/static/js/main.js where similar code appears ot
+    // make the menu work on the wiki domain
+    for (let menu of document.querySelectorAll('select.autosubmit')) {
```

Comment:
Might it be a little better to do the query once and save the result in a variable, instead of doing the query each time we go through the `for` loop?

```suggestion
    let langSelector = document.querySelectorAll('select.autosubmit');
    for (let menu of langSelector) {
```

Then again, looking at the HTML, could we do:

```suggestion
    let langSelector = document.getElementById('language');
    langSelector.addEventListener('change', function() {
        this.form.submit();
    });
```

Reply:
Surely `for (let menu of document.querySelectorAll('select.autosubmit')) {` only executes the `document.querySelectorAll('select.autosubmit')` call just once, no?

Reply:
Sure, but it seems like we do not even need a `for` loop.

Reply:
The jQuery version of the code doesn't need a for loop because jQuery magically handles multiple items as if they were one item. As far as I can tell, there is only one 'select.autosubmit' element on the page, but I didn't want to rule out the possibility that we might have them elsewhere as well. But I could switch to querySelector() instead of querySelectorAll() and drop the loop.

But the loop is efficient as it is written. The querySelectorAll() call returns an iterable object, and the for/of loop then iterates that object. querySelectorAll() itself is only called once.